simuq/aais/qmachine_factory.py

```diff
@@ -0,0 +1,9 @@
+from abc import ABC, abstractmethod
+from simuq.qmachine import QMachine
+
+class QMachineFactory(ABC):
+    @staticmethod
+    @abstractmethod
+    def generate_qmachine(*args, **kwargs) -> QMachine:
```

Comment:
Looking at the implementations a bit more, it might actually be better to just have each of the AAIS implementations be standalone functions (still better than standalone _variables_), since the signature of `generate_qmachine` takes arbitrary arguments. I'd see more value in this interface if there was more uniformity in what gets passed in.